library/ecdh.c

```diff
@@ -87,7 +87,9 @@ int mbedtls_ecdh_compute_shared( mbedtls_ecp_group *grp, mbedtls_mpi *z,
  */
 void mbedtls_ecdh_init( mbedtls_ecdh_context *ctx )
 {
-    memset( ctx, 0, sizeof( mbedtls_ecdh_context ) );
+    if ( ctx != NULL ) {
```

Comment:
Is there a good reason to introduce this kind of null check? The function `mbedtls_ecdh_init` expects a pointer to a context as an argument, not a nullable pointer. Ignoring null pointers tends to make errors harder to debug because they are not noticed until after the point where the error happens. It is generally better to let the program crash as soon as possible, rather than later.